src/Microsoft.SqlTools.ServiceLayer/SqlContext/SqlToolsSettings.cs

```diff
@@ -103,19 +144,116 @@ public bool IsIntelliSenseEnabled
         }
     }
 
+    public interface ISqlToolsSettingsValues
+    {
+        IntelliSenseSettings IntelliSense { get; set; }
+        QueryExecutionSettings QueryExecutionSettings { get; set; }
+        FormatterSettings Format { get; set; }
+        ObjectExplorerSettings ObjectExplorer { get; set; }
+    }
+
+    public class CompoundToolsSettingsValues: ISqlToolsSettingsValues
```

Comment:
I think these classes need their own files. It's starting to get cluttered in here.